docs/site/content/docs/latest/storage.md

```diff
@@ -0,0 +1,146 @@
+# Create Persistent Volumes with Storage Classes
+
+This topic explains how to use dynamic storage in workload clusters.
+
+## Overview: PersistentVolume, PersistentVolumeClaim, and StorageClass
+
+Within a Kubernetes cluster, `PersistentVolume` (PV) objects provide shared storage for cluster pods that is unaffected by pod lifecycles.
+Storage is provisioned to the PV through a `PersistentVolumeClaim` (PVC) object, which defines how much and how the pod accesses the underlying storage.
+For more information, see [Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) in the Kubernetes documentation.
+
+Cluster administrators can define `StorageClass` objects that let cluster users dynamically create PVC and PV objects with different storage types and rules.
+Tanzu also provides default `StorageClass` objects that let users provision persistent storage in a turnkey environment.
+
+`StorageClass` objects include a `provisioner` field identifying the internal or external service plug-in that provisions PVs, and a `parameters` field that associates the Kubernetes storage class with storage options defined at the infrastructure level, such as VM Storage Policies in vSphere.
+For more information, see [Storage Classes](https://kubernetes.io/docs/concepts/storage/storage-classes/) in the Kubernetes documentation.
+
+## Supported Storage Types
+
+Tanzu Kubernetes Grid supports `StorageClass` objects for different storage types, provisioned by Kubernetes internal ("in-tree") or external ("out-of-tree") plug-ins.
+
+### Storage Types
+
+- vSphere Cloud Native Storage (CNS)
+- Amazon EBS
+- Azure Disk
+- iSCSI
+- NFS
+
+See [Default Storage Classes](#defaults) below for vSphere CNS, Azure EBS, and Azure Disk default storage classes.
+
+### Plug-in Locations
+
+- Kubernetes internal ("in-tree") storage.
+      - Ships with core Kubernetes; `provider` values are prefixed with `kubernetes.io`, e.g. `kubernetes.io/aws-ebs`.
+- External ("out-of-tree") storage.
+      - Can be anywhere defined by `provider` value, e.g. `csi.vsphere.vmware.com`.
+      - Follow the Container Storage Interface (CSI) standard for external storage.
+
+## Default Storage Classes
+
+Tanzu provides default `StorageClass` objects that let workload cluster users provision persistent storage on their infrastructure in a turnkey environment, without needing `StorageClass` objects created by a cluster administrator.
+
+The `ENABLE_DEFAULT_STORAGE_CLASS` variable is set to `true` by default in the cluster configuration file passed to `--file` option of `tanzu cluster create`, to enable the default storage class for a workload cluster.
+
+The Tanzu default storage class definitions are:
+
+### vSphere CNS
+
+```sh
```

Comment:
I believe this should be `yaml` instead of `sh`.

Reply:
changed